src/topology/sequences.lean

```diff
@@ -150,54 +149,39 @@ iff.intro
         show f p ∈ A, from
           mem_of_is_closed_sequential ‹is_closed A› ‹∀ n, f (x n) ∈ A› ‹(f∘x ⟶ f p)›)
 
-
 end topological_space
 
-/- Statements about sequences in metric spaces -/
-namespace metric
-variable [metric_space α]
-variables {ε : ℝ}
+namespace topological_space
+
+namespace first_countable_topology
 
--- necessary for the next instance
-set_option eqn_compiler.zeta true
-/-- Show that every metric space is sequential. -/
-instance : sequential_space α :=
+/-- Every first-countable space is sequential. -/
+instance [topological_space α] [first_countable_topology α] : sequential_space α :=
 ⟨show ∀ M, sequential_closure M = closure M, from assume M,
   suffices closure M ⊆ sequential_closure M,
     from set.subset.antisymm (sequential_closure_subset_closure M) this,
-  assume (p : α) (_ : p ∈ closure M),
-  -- we construct a sequence in α, with values in M, that converges to p
-  -- the first step is to use (p ∈ closure M) ↔ "all nhds of p contain elements of M" on metric
-  -- balls
-  have ∀ n : ℕ, ball p ((1:ℝ)/((n+1):ℝ)) ∩ M ≠ ∅ := assume n : ℕ,
-    mem_closure_iff.mp ‹p ∈ (closure M)› (ball p ((1:ℝ)/((n+1):ℝ))) (is_open_ball)
-    (mem_ball_self $ one_div_pos_of_pos $ add_pos_of_nonneg_of_pos (nat.cast_nonneg n) zero_lt_one),
-
-  -- from this, construct a "sequence of hypothesis" h, (h n) := _ ∈ {x // x ∈ ball (1/n+1) p ∩ M}
-  let h := λ n : ℕ, (classical.indefinite_description _ (set.exists_mem_of_ne_empty (this n))),
-  -- and the actual sequence
-      x := λ n : ℕ, (h n).val in
-
-  -- now we construct the promised sequence and show the claim
-  show ∃ x : ℕ → α, (∀ n : ℕ, ((x n) ∈ M)) ∧ (x ⟶ p), from
-    ⟨x, assume n,
-        have (x n) ∈ ball p ((1:ℝ)/((n+1):ℝ)) ∩ M := (h n).property, this.2,
-        suffices ∀ ε > 0, ∃ n0 : ℕ, ∀ n ≥ n0, dist (x n) p < ε,
-          by simpa only [metric.tendsto_at_top],
-        assume ε _,
-        -- we apply that 1/n converges to zero to the fact that (x n) ∈ ball p ε
-        have ∀ ε > 0, ∃ n0 : ℕ, ∀ n ≥ n0, dist (1 / (↑n + 1)) (0:ℝ) < ε :=
-          metric.tendsto_at_top.mp tendsto_one_div_add_at_top_nhds_0_nat,
-        let ⟨n0, hn0⟩ := this ε ‹ε > 0› in
-        show ∃ n0 : ℕ, ∀ n ≥ n0, dist (x n) p < ε, from
-          ⟨n0, assume n ngtn0,
-            calc dist (x n) p < (1:ℝ)/↑(n+1) : (h n).property.1
-              ... = abs ((1:ℝ)/↑(n+1)) :
-                eq.symm $ abs_of_nonneg $ div_nonneg' zero_le_one $ nat.cast_nonneg _
-              ... = abs ((1:ℝ)/↑(n+1) - 0) : by simp
-              ... = dist ((1:ℝ)/↑(n+1)) 0 : eq.symm $ real.dist_eq ((1:ℝ)/↑(n+1)) 0
-              ... < ε : hn0 n ‹n ≥ n0›⟩⟩⟩
-
-set_option eqn_compiler.zeta false
-
-end metric
+  assume (p : α) (hp : p ∈ closure M),
+  exists.elim (mono_seq_of_has_countable_basis _
+    (nhds_generated_countable p)) $ assume g ⟨gmon, gbasis⟩,
+  have x : ∀ i, (g i ∩ M : set α), {
```

Comment:
Trying to read the diff without starting Lean, I have a hard time understanding what this line says: it should say something about the set `g i ∩ M`, but I guess the property you are proving is the result of the proof block below. It would help a lot if you would say what you are proving here. Also, minor nitpick, the mathlib style is not
```lean
... {
  proof 
  block
},
```
but instead
```lean
...
{ proof 
  block },
```

Reply:
Ok, I fixed the braces style.

This line indeed proves something about the set `g i ∩ M`, namely that `∀ i, g i ∩ M`, which is short for `∀ i, ↑(g i ∩ M)` and is equivalent to `∀ i, (g i ∩ M) ≠ ∅`.  I'll also add comments to the proof.